app/actions/local/channel.ts

```diff
@@ -486,3 +491,28 @@ export async function showUnreadChannelsOnly(serverUrl: string, onlyUnreads: boo
         prepareRecordsOnly: false,
     });
 }
+
+export const updateDmGmDisplayName = async (serverUrl: string) => {
+    const database = DatabaseManager.serverDatabases[serverUrl]?.database;
+    if (!database) {
+        return {error: `${serverUrl} database not found`};
+    }
+
+    try {
+        const currentUserId = await getCurrentUserId(database);
+        if (!currentUserId) {
+            return {error: 'The current user id could not be retrieved from the database'};
+        }
+
+        const channels = await queryUserChannelsByTypes(database, currentUserId, ['G', 'D']).fetch();
+        const userIds = channels.map((ch) => getUserIdFromChannelName(currentUserId, ch.name)).filter((uid) => uid !== undefined);
```

Comment:
if you change this to a reduce, you can make it O(1) instead of O(1) + O(1)

Reply:
refactored